images/cfw/opt/x1plus/lib/python/x1plus/services/x1plusd/expansion/i2c.py

```diff
@@ -21,13 +24,32 @@ def __init__(self, daemon, config, ftdi_path, port_name):
         self.daemon = daemon
         self.devices = []
         
+        self.drivers = {}
+        
+        # Load all valid driver classes from directory
+        for filename in os.listdir(self.DRIVER_DIR):
+            if filename.endswith('.py'):
+                module_name = filename[:-3]
+                driver_path = os.path.join(self.DRIVER_DIR, filename)
+                spec = importlib.util.spec_from_file_location(module_name, driver_path)
+                module = importlib.util.module_from_spec(spec)
+                spec.loader.exec_module(module)
+                
+                for name, driver in inspect.getmembers(module, inspect.isclass):
+                    if hasattr(driver, 'device_type'): 
+                        self.DEVICE_DRIVERS[driver.device_type] = driver
```

Comment:
TODO later - after loading all drivers, and mapping all devices, could delete unused drivers from DEVICE_DRIVERS to remove the class references from memory. Unless this does not seem necessary?